public/apps/configuration/panels/permission-list.tsx

```diff
@@ -0,0 +1,242 @@
+  EuiFlexItem,
+  EuiInMemoryTable,
+  EuiLink,
+  EuiPageBody,
+  EuiPageContent,
+  EuiPageContentHeader,
+  EuiPageContentHeaderSection,
+  EuiPageHeader,
+  EuiPopover,
+  EuiText,
+  EuiTitle,
+  EuiIcon,
+} from '@elastic/eui';
+import { difference } from 'lodash';
+import React, { useEffect, useState } from 'react';
+import { AppDependencies } from '../../types';
+import { ActionGroupListingItem, fetchActionGroupListing } from '../utils/action-groups-utils';
+import { renderCustomization } from '../utils/display-utils';
+import { requestDeleteUsers } from '../utils/internal-user-list-utils';
+
+
+function renderBooleanToCheckMark(value: boolean): React.ReactNode {
+  return value ? <EuiIcon type="check" /> : ''
+}
+
+const COLUMNS = [
+  {
+    field: 'name',
+    name: 'name',
+    sortable: true,
+  },
+  {
+    field: 'type',
+    name: 'Type',
+    sortable: true,
+  },
+  {
+    field: 'hasClusterPermission',
+    name: 'Cluster permission',
+    render: renderBooleanToCheckMark,
+  },
+  {
+    field: 'hasIndexPermission',
+    name: 'Index permission',
+    render: renderBooleanToCheckMark,
+  },
+  {
+    field: 'reserved',
+    name: 'Customization',
+    render: renderCustomization,
+  },
+];
+
+export function PermissionList(props: AppDependencies) {
+  const [actionGroups, setActionGroups] = useState<ActionGroupListingItem[]>([]);
+  const [errorFlag, setErrorFlag] = useState(false);
+  const [selection, setSelection] = useState<ActionGroupListingItem[]>([]);
+  const [isActionsPopoverOpen, setActionsPopoverOpen] = useState(false);
+  const [isCreateActionGroupPopoverOpen, setCreateActionGroupPopoverOpen] = useState(false);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        setActionGroups(await fetchActionGroupListing(props.coreStart.http));
+      } catch (e) {
+        console.log(e);
+        setErrorFlag(true);
+      }
+    };
+
+    fetchData();
+  }, [props.coreStart.http]);
+
+  const handleDelete = async () => {
+    const usersToDelete: string[] = selection.map((r) => r.name);
+    try {
+      await requestDeleteUsers(props.coreStart.http, usersToDelete);
+      setActionGroups(difference(actionGroups, selection));
+      setSelection([]);
+    } catch (e) {
+      console.log(e);
+    } finally {
+      setActionsPopoverOpen(false);
+    }
+  };
+
+  const actionsMenuItems = [
+    <EuiContextMenuItem
+      key="edit"
+      // TODO edit action
+      onClick={() => {}}
+      disabled={selection.length !== 1}
+    >
+      Edit
+    </EuiContextMenuItem>,
+    <EuiContextMenuItem
+      key="duplicate"
+      // TODO duplicate action
+      onClick={() => {}}
+      disabled={selection.length !== 1}
+    >
+      Duplicate
+    </EuiContextMenuItem>,
+    <EuiContextMenuItem key="delete" onClick={handleDelete} disabled={selection.length === 0}>
+      Delete
+    </EuiContextMenuItem>,
+  ];
+
+  const createActionGroupMenuItems = [
+    <EuiContextMenuItem
+      key="blank"
```

Comment:
This is same as Line 138. Probably we should give it a meaningful key name

Reply:
Sure, updated.